### PR TITLE
Clarify pointer arguments in some daemons

### PR DESCRIPTION
**Pull request by kgaillot**
Mark pointer arguments as const where appropriate, and ensure doxygen blocks use [in] or [in,out] as appropriate, in functions for pacemakerd, pacemaker-based, and pacemaker-fenced.

This is not comprehensive but focuses on functions that have doxygen blocks and some functions called by them.

### CODE REVIEW

**Comment by kgaillot** [comment]
@nrwahl2 , here's another low-priority follow-up

File: daemons/fenced/fenced_commands.c [hunk 1]
```diff
@@ -974,7 +975,7 @@ map_action(GHashTable *params, const char *action, const char *value)
  * \brief Create device parameter table from XML
  *
  * \param[in]     name    Device name (used for logging only)
- * \param[in,out] params  Device parameters
+ * \param[in,out] dev     XML containing device parameters
```
Comment (nrwahl2) [inline daemons/fenced/fenced_commands.c:978]: Can this just be `[in]`? `xml2list()` is const and the result is a hash table of copied name-value strings.

Comment (kgaillot) [reply daemons/fenced/fenced_commands.c:978]: Yep, not sure how I missed that

Comment (kgaillot) [reply daemons/fenced/fenced_commands.c:978]: Ah, it was already marked in,out, and I assumed that was correct

File: daemons/fenced/fenced_commands.c [hunk 2]
```diff
@@ -3541,11 +3546,11 @@ handle_reply(pcmk__client_t *client, xmlNode *request, const char *remote_peer)
  * \internal
  * \brief Handle a message from an IPC client or CPG peer
  *
- * \param[in] client      If not NULL, IPC client that sent message
- * \param[in] id          If from IPC client, IPC message ID
- * \param[in] flags       Message flags
- * \param[in] message     Message XML
- * \param[in] remote_peer If not NULL, CPG peer that sent message
+ * \param[in,out] client      If not NULL, IPC client that sent message
```
Comment (nrwahl2) [inline daemons/fenced/fenced_commands.c:3549]: `client` can be made const if you do the same for `handle_reply()`. All `handle_reply()` does is null-check and get the name.

Comment (nrwahl2) [reply daemons/fenced/fenced_commands.c:3549]: Sorry, missed the assignment to `.ipc_client`

Comment (kgaillot) [reply daemons/fenced/fenced_commands.c:3549]: It's non-const when assigned to pcmk__request_t, which needs it that way

File: daemons/fenced/fenced_commands.c (hunk 1 above)
Comment (nrwahl2) [reply daemons/fenced/fenced_commands.c:978]: Looks good to me otherwise

Comment (kgaillot) [reply daemons/fenced/fenced_commands.c]: Rebased on current main and updated this, no other changes

**Pull request merged**